frame/staking/src/tests.rs

```diff
@@ -1744,32 +1740,33 @@ fn switching_roles() {
 }
 
 #[test]
-fn wrong_vote_is_null() {
+fn wrong_vote_is_moot() {
 	ExtBuilder::default()
-		.nominate(false)
-		.validator_pool(true)
+		.add_staker(
+			61,
+			60,
+			500,
+			StakerStatus::Nominator(vec![
+				11, 21, // good votes
+				1, 2, 15, 1000, 25, // crap votes. No effect.
+			]),
+		)
 		.build_and_execute(|| {
-			assert_eq_uvec!(validator_controllers(), vec![40, 30]);
-
-			// put some money in account that we'll use.
-			for i in 1..3 {
-				let _ = Balances::deposit_creating(&i, 5000);
-			}
-
-			// add 1 nominators
-			assert_ok!(Staking::bond(Origin::signed(1), 2, 2000, RewardDestination::default()));
-			assert_ok!(Staking::nominate(
-				Origin::signed(2),
-				vec![
-					11, 21, // good votes
-					1, 2, 15, 1000, 25 // crap votes. No effect.
-				]
-			));
-
-			// new block
+			// the genesis validators already reflect the above vote, nonetheless start a new era.
 			mock::start_active_era(1);
 
+			// new validators
 			assert_eq_uvec!(validator_controllers(), vec![20, 10]);
+
+			// our new voter is taken into account
+			assert!(Staking::eras_stakers(Staking::active_era().unwrap().index, 11)
```

Comment:
If we are adding new code I think we should use `mock::active_era()` instead of `Staking::active_era().unwrap().index`